### DISCUSSION TITLE
running Sloeber on a Raspi

### DISCUSSION THREAD

**Discussion by PerennialNovice** [Sloeber/arduino-eclipse-plugin#1449] [Q&A] [answered]
Is it possible to run Sloeber on a Raspberry PI 4?
Does not work "out of the box"

**Comment by jantje** [comment]
I have no clue but:.
Does CDT work on that platform?
Does eclipse work on that platform?
Does Arduino IDE work on that platform?

If any of these answers is no; it is not likely Sloeber will work on that platform.

**Comment by PerennialNovice** [comment]
with a bit of fiddling and cursing it works :)
had to go the way of 1st installing CDT and adding Sloeber via marketplace
performance is not really good on a RasPi 4 8Gb but it works

**Reply by jantje** [reply]
>performance is not really good on a RasPi 4 8Gb but it works

That is as I would expect. You are after all running a high end development tool on a low end machine.

>with a bit of fiddling and cursing it works :)

Well happy to see there is a way to get it to work.

**Comment by PerennialNovice** [comment, marked as answer]
Hi jantje,

here is a (very) quick writeup of the steps the led to success for me:

- install [latest eclipse CDT](https://www.eclipse.org/cdt/downloads.php)

- proceed according to the [wokaround of issue](https://github.com/Sloeber/arduino-eclipse-plugin/issues/1361#issuecomment-916948294) #1361

- I then ran into a timeout when sloeber tries to download avr board package. So I [downloaded manually](http://downloads.arduino.cc/tools/avr-gcc-7.3.0-atmel3.6.1-arduino7-aarch64-pc-linux-gnu.tar.bz2) and copied it to `[EclipseInstallationFolder]/arduinoPlugin/downloads`

- next thing (boards not parsing) was that, despite Java 11 beeing installed on the system (I don't know whether it came pre-installed because I actively downloaded and installed OpenJDK at the first attempt of installing and running sloeber product), I got error messages that could be identified (reading issue #1403) as related to a too recent Java version. That can be fixed by [modifying eclipse.ini / sloeber.ini](https://github.com/Sloeber/arduino-eclipse-plugin/issues/1403#issuecomment-962290607) to us the local Java VM (`/usr/lib/jvm/java-11-openjdk-arm64/bin` for my setup)

I think that is about it what it took to get Sloeber running on an Raspberry Pi 4

maybe it is of use to someone sometime :)
